### PR TITLE
Feat: Added helper functions

### PR DESCRIPTION
**Pull request by nishasen**
Added helper functions these functions basically contain the data that are getting repeated in the component. So kept it here and mapping it in components. Also it contains calculations of total price of the cart items.

### CODE REVIEW

**Review by sadathshariff — APPROVED** [review]
Looks Good, like how you have broken it down into smaller functions

File: diagon-alley/src/GeneralFunctions/FilterOutDetail.js [hunk 1]
```diff
@@ -0,0 +1,34 @@
+export const FilterOutDetail = (onlyInStock, fastDelivery, ratingAbove4, newProduct) =>
+        [{
+            fieldLabel:"Exclude out of stock", 
+            fieldForLabel:"excludeOutOfStock",
+            fieldName:"EXCLUDE_OUT_OF_STOCK",
+            fieldType:"checkbox",
+            fieldValue:"excludeOutOfStock",
+            fieldFilter: onlyInStock,
+            fieldLabelClassName:"text-white",
```
Comment (ttahm3d) [inline diagon-alley/src/GeneralFunctions/FilterOutDetail.js]: Don't give extra context. What i mean is you knoe eveything is related to field, then why prefix everything with field?

**Review by ttahm3d — COMMENTED** [review]
Everything looks good. One suggestion is tho avoid giving extra context to already known things

File: diagon-alley/src/GeneralFunctions/FilterOutDetail.js (hunk 1 above)
Comment (nishasen) [reply diagon-alley/src/GeneralFunctions/FilterOutDetail.js]: Yes got it will change this

**Pull request merged**